File: Python3/Medium/WrongAnswer/average-waiting-time_0.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        
        start = customers[0][0]
        end = start + customers[0][1]
        total = end - start
        for i in range(1, len(customers)):
            start = max(end, customers[i][0])
            end = start + customers[i][1]
            total += end - start
        return total / len(customers)

File: Java/Medium/WrongAnswer/average-waiting-time_0.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        
        int n = customers.length;
        int start = customers[0][0];
        int end = start + customers[0][1];
        int total = end - start;
        for (int i = 1; i < n; i++) {
            int[] c = customers[i];
            start = Math.max(start, c[0]);
            end = start + c[1];
            total += end - c[0];
            start = end;
        }
        return (double) total / n;
    }
}